controllers/user/login.js

```diff
@@ -44,33 +58,63 @@ router
         });
         renderForm(req, res);
     })
-    .post((req, res, next) => {
+    .post(async (req, res, next) => {
         logger.info('Login attempted');
-        passport.authenticate('local', function(err, user) {
-            if (err) {
-                logger.error('Authentication failed', err);
-                next(err);
-            } else if (user) {
-                req.logIn(user, function(loginErr) {
-                    if (loginErr) {
-                        logger.error('Login failed', loginErr);
-                        next(loginErr);
-                    } else {
-                        logger.info('Login succeeded');
-                        redirectUrlWithFallback(req, res, '/user');
+
+        // Key all rate-limited login attempts by the user's IP and the attempted username
+        const uniqueKey = `${req.ip}_${req.body.username}`;
```

Comment:
As this is a fairly important key could we name this something like `RATE_LIMIT_UNIQUE_KEY`